File: 3296_Minimum_Number_of_Seconds_to_Make_Mountain_Height_Zero/solution.py
import math
class Solution:
    def minNumberOfSeconds(self, mountainHeight: int, workerTimes: List[int]) -> int:
        lo, hi = 1, 10**16
        while lo < hi:
            m = (lo + hi) >> 1
            s = sum(int(math.sqrt(m / t * 2 + 0.25) - 0.5) for t in workerTimes)
            if s >= mountainHeight: hi = m
            else: lo = m + 1
        return lo

File: 3296_Minimum_Number_of_Seconds_to_Make_Mountain_Height_Zero/solution.c
long long minNumberOfSeconds(int mountainHeight, int* workerTimes, int n) {
    long long lo = 1, hi = 1e16;
    while (lo < hi) {
        long long m = (lo + hi) >> 1, s = 0;
        for (int i = 0; i < n; i++) s += (long long)(sqrt(m / (double)workerTimes[i] * 2 + .25) - .5);
        if (s >= mountainHeight) hi = m;
        else lo = m + 1;
    }
    return lo;
}